samples-android/ButtonClicker/jni/ButtonClickerNativeActivity.cpp

```diff
@@ -105,15 +105,14 @@ void Engine::InitGooglePlayGameServices() {
  *
  */
 void Engine::OnAuthActionStarted(gpg::AuthOperation op) {
+  startup_mutex_.lock();
   if (!initialized_resources_) return;
+
   ndk_helper::JNIHelper::GetInstance()->RunOnUiThread([this, op]() {
     EnableUI(false);
-    authorizing_ = true;
     if (op == gpg::AuthOperation::SIGN_IN) {
-      LOGI("Signing in to GPG");
```

Comment:
can you keep the log messages?  They help diagnose issues which there always are with authentication.